server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -532,4 +575,50 @@ public void testNoSearchIdleForAnyReplicaCount() throws ExecutionException, Inte
         indexShard = getIndexShard(replicaShardNode);
         assertFalse(indexShard.isSearchIdleSupported());
     }
+
+    public void testFallbackToNodeToNodeSegmentCopy() throws Exception {
```

Comment:
This test will fail without the change made in `PeerRecoveryTargetService` as part of this PR.